Automattic/newspack-blocks
# feat(modal-checkout): update modal checkout checkout form with newspack ui

### All Submissions:






### Changes proposed in this Pull Request:



Part of `1206197591054328/1206405274735557/f`

This PR updates the remainder of the checkout form with newspack ui including:

The billing info form:
![Screenshot 2024-01-30 at 17 15 39](https://github.com/Automattic/newspack-blocks/assets/17905991/78ddcaf4-98b7-4369-96df-2e5b84c7da3b)

The checkout form:
![Screenshot 2024-01-30 at 17 17 29](https://github.com/Automattic/newspack-blocks/assets/17905991/d5a4f65b-5da2-4fa8-9778-2c28f0cece0e)

The variation selection form:
![Screenshot 2024-01-30 at 17 19 07](https://github.com/Automattic/newspack-blocks/assets/17905991/a38210f9-2608-47ff-96f5-c9b47ea100a5)

### How to test the changes in this Pull Request:

Important! Before testing, make sure you have https://github.com/Automattic/newspack-plugin/pull/2878 checked out as many of the woocommerce form styles have been overridden there.

1. On your test site, add a donation block, a checkout button block for variation product, and another checkout button block for a subscription product to any page.
2. Also ensure you've enabled coupons in WooCommerce settings AND have installed [WooCommerce Subscriptions Gifting](https://github.com/woocommerce/woocommerce-subscriptions-gifting)
3. On the front end, visit this page and select one of the checkout buttons of the donation block.
4. Confirm you see the updated ui in the billing and checkout portions of the checkout flow (pictured above)
5. Repeat the above steps for the checkout block for the variable product.
6. Confirm you see the updated ui in the variation, billing, and checkout portions of the checkout flow (pictured above). Lookout for the coupon field in the checkout portion of the flow:
![Screenshot 2024-01-30 at 17 28 47](https://github.com/Automattic/newspack-blocks/assets/17905991/13f98d5e-7b11-4bd7-99c5-6e67c6e78a66)
8. Repeat again for the checkout block for the subscription product.
9. Confirm you see the updated ui in the billing and checkout portions of the checkout flow (pictured above). Look out for the subscription gift toggle in the billing info section of the flow:
![Screenshot 2024-01-30 at 17 28 37](https://github.com/Automattic/newspack-blocks/assets/17905991/69dbf828-f2da-419d-978e-5fd9efdece14)
10. Make sure to smoke test the form, submitting with missing or invalid inputs, etc.

**Testing the standalone modal checkout**

We also want to make sure the checkout block doesn't appear too broken when newspack ui is not present. We don't need it to match, but the checkout flow should use a combination of default WC and a bit of custom newspack block styles.

1. Check out `master` over in the `newspack-plugin` repo so newspack ui is no longer present.
2. Go through the testing instructions above and make sure nothing looks janky 😰 

### Other information:





---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206405274735557
